app/assets/javascripts/admin/services/bulk_products.js.coffee

```diff
@@ -16,9 +16,9 @@ angular.module("ofn.admin").factory "BulkProducts", (PagedFetcher, dataFetcher,
           @unpackProduct newProduct
           @insertProductAfter(product, newProduct)
 
-    updateVariantLists: (serverProducts, productsWithUnsavedVariants) ->
-      for product in productsWithUnsavedVariants
-        server_product = @findProductInList(product.id, serverProducts)
+    updateVariantLists: (serverProducts) ->
+      for server_product in serverProducts
+        product = @findProductInList(server_product.id, @products)
```

Comment:
Hi @ltrls, this means that we will be iterating through every single product, rather than just those that have been changed. Is there any particular reason why this is necessary?